File: app/src/main/java/com/renanferrari/testeandroidv2/application/di/modules/ActivityModule.java
package com.renanferrari.testeandroidv2.application.di.modules;

import com.renanferrari.testeandroidv2.application.di.scopes.ActivityScope;
import com.renanferrari.testeandroidv2.application.ui.login.LoginActivity;
import com.renanferrari.testeandroidv2.application.ui.statements.StatementsActivity;
import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module public abstract class ActivityModule {
  @ActivityScope
  @ContributesAndroidInjector
  abstract LoginActivity contributeLoginActivity();

  @ActivityScope
  @ContributesAndroidInjector
  abstract StatementsActivity contributeStatementsActivity();
}

File: BankApp/app/src/main/java/br/com/rms/bankapp/di/module/ActivityModule.kt
package br.com.rms.bankapp.di.module

import br.com.rms.bankapp.di.ActivityScope
import br.com.rms.bankapp.ui.home.HomeActivity
import br.com.rms.bankapp.ui.home.HomeFragmentProvider
import br.com.rms.bankapp.ui.login.LoginActivity
import br.com.rms.bankapp.ui.login.LoginFragmentProvider
import br.com.rms.bankapp.ui.splash.SplashActivity
import dagger.Module
import dagger.android.ContributesAndroidInjector


@Module
abstract class ActivityModule {

    @ActivityScope
    @ContributesAndroidInjector()
    internal abstract fun contributeSplashActivity(): SplashActivity


    @ActivityScope
    @ContributesAndroidInjector(modules = [LoginFragmentProvider::class])
    internal abstract fun contributeLoginActivity(): LoginActivity

    @ActivityScope
    @ContributesAndroidInjector(modules = [HomeFragmentProvider::class])
    internal abstract fun contributeHomeActivity(): HomeActivity

}